### DISCUSSION TITLE
Close Modal - Jetstream

### DISCUSSION THREAD

**Discussion by Abdallah-Tah** [livewire/livewire#5399] [1. Help]
Hello,

How do I close a modal window after submitting while it is processing the request, right now it will close after the data has been stored. Therefore, I am looking to close the modal and run a loading page while it is processing.

Thank you,

**Comment by alfakiat** [comment]
Try to use `emit` to close Modal in your function before execute your code
like this in your livewire:

ExampleComponent.php:
```
public function store()
{
    $this->emit('closeMyModal');
    // here your code ...
}
```

**Reply by lrljoe** [reply]
Yep, or use the loading states to look after this for you and target a specific action.
